packages/assets-controllers/src/TokenDetectionController.test.ts

```diff
@@ -256,29 +260,10 @@ describe('TokenDetectionController', () => {
     });
   });
 
-  it('should detect supported networks correctly', () => {
-    tokenDetection.configure({
-      chainId: SupportedTokenDetectionNetworks.mainnet,
-    });
-
-    expect(
-      isTokenDetectionSupportedForNetwork(tokenDetection.config.chainId),
-    ).toBe(true);
-    tokenDetection.configure({ chainId: SupportedTokenDetectionNetworks.bsc });
-    expect(
-      isTokenDetectionSupportedForNetwork(tokenDetection.config.chainId),
-    ).toBe(true);
-    tokenDetection.configure({ chainId: ChainId.goerli });
-    expect(
-      isTokenDetectionSupportedForNetwork(tokenDetection.config.chainId),
-    ).toBe(false);
-  });
-
   it('should not autodetect while not on supported networks', async () => {
-    tokenDetection.configure({
-      selectedAddress: '0x1',
-      chainId: ChainId.goerli,
-      isDetectionEnabledForNetwork: false,
+    await tokenDetection.detectTokens({
```

Comment:
Hmm... we're detecting tokens before we start. While the test obviously works, I feel like we're cheating a bit here (and it makes the test harder to follow).

Is it possible to extract the creation of the TokenDetectionController to a separate function and then call it manually here?

Reply:
Fixed here: 4d58817181d2b4ae3cc7aff7d2a7592e0038b375